uxarray/grid/utils.py

```diff
@@ -320,3 +320,74 @@ def _get_lonlat_rad_face_edge_nodes(
     face_edges_lonlat_rad[valid_mask, 1] = node_lat_rad[valid_edges]
 
     return face_edges_lonlat_rad.reshape(n_face, n_max_face_edges, 2, 2)
+
+
+@njit(cache=True)
+def _edge_nodes_cartesian_kernel(
+    face_idx, face_node_connectivity, n_edges_per_face, node_x, node_y, node_z
+):
+    """Computes the Cartesian Coordinates of the edges that make up a given face."""
+    # Number non-fill-value edges
+    n_edges = n_edges_per_face[face_idx]
+
+    # Allocate data for face_edge_nodes
+    face_edge_nodes = np.empty((n_edges, 2, 3), dtype=np.float64)
+
+    start_nodes = face_node_connectivity[face_idx, 0:n_edges]
+    end_nodes = np.roll(start_nodes, -1)
+
+    # Assign x coordinates of start and end nodes
+    face_edge_nodes[0:n_edges, 0, 0] = node_x[start_nodes]
+    face_edge_nodes[0:n_edges, 1, 0] = node_x[end_nodes]
+
+    # Assign y coordinates of start and end nodes
+    face_edge_nodes[0:n_edges, 0, 1] = node_y[start_nodes]
+    face_edge_nodes[0:n_edges, 1, 1] = node_y[end_nodes]
+
+    # Assign z coordinates of start and end nodes
+    face_edge_nodes[0:n_edges, 0, 2] = node_z[start_nodes]
+    face_edge_nodes[0:n_edges, 1, 2] = node_z[end_nodes]
+
+    return face_edge_nodes
+
+
+@njit(cache=True)
+def _edge_nodes_spherical_kernel(
```

Comment:
Similarly replace all calls from `_get_lonlat_rad_face_edge_nodes` to `_edge_nodes_spherical_kernel`